### ISSUE TITLE
🐛 False positive linter diagnostic on GritQL files

### ISSUE THREAD

**Issue by notramo** [biomejs/biome#5638]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm
  JS_RUNTIME_VERSION:           v22.6.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.10

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /redacted-path-to-project-dir/biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

The `--write` option causes a false positive.

`bun run biome check --write extension.grit` - this exits with an error message: `no diagnostic message provided`
`bun run biome check extension.grit` - it shows no error when the `--write` option is not specified

### Expected result

`--write` should not affect the diagnostics, especially that it's a false positive.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Closed by https://github.com/biomejs/biome/pull/5611
